compiler/rustc_typeck/src/check/fn_ctxt/checks.rs

```diff
@@ -291,7 +324,42 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             } else {
                 false
             };
-            param_count_error(expected_arg_count, supplied_arg_count, "E0061", false, sugg_unit);
+
+            // are we passing elements of a tuple without the tuple parentheses?
+            let chosen_arg_tys = match &expected_arg_tys[..] {
+                [] => fn_inputs,
+                args @ [..] => args,
+            };
```

Comment:
This match feels a bit complex.

```suggestion
            let chosen_arg_tys = if expected_arg_tys.is_empty() {
                fn_inputs
            } else {
                expected_arg_tys
            };
```

Reply:
I also don't really understand what this code is doing. Could you add a comment perhaps explaining why we have to check if `expected_arg_tys` is empty and if so use `fn_inputs` instead?

Reply:
Thanks, and sure - comment updated.